### PR TITLE
resource_manager: replace MaxToken by BurstLimit

### PR DESCRIPTION
**Pull request by CabinfeverB**
Signed-off-by: Cabinfever_B <cabinfeveroier@gmail.com>

### What problem does this PR solve?

Issue Number: ref #5851

### What is changed and how does it work?
1. replace MaxToken by BurstLimit
2. split TokenBucket into TokenLimitSettings and GroupTokenBucketState in GroupTokenBucket

```commit-message
```

### Check List

Tests

- Unit test
- Integration test

### Release note

```release-note
None.
```

### CODE REVIEW

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by nolouch** [comment]
ptal @JmPotato

File: pkg/mcs/resource_manager/server/resource_group.go [hunk 1]
```diff
@@ -167,7 +167,7 @@ func FromProtoResourceGroup(group *rmpb.ResourceGroup) *ResourceGroup {
 func (rg *ResourceGroup) RequestRU(now time.Time, neededTokens float64, targetPeriodMs uint64) *rmpb.GrantedRUTokenBucket {
 	rg.Lock()
 	defer rg.Unlock()
-	if rg.RUSettings == nil {
+	if rg.RUSettings == nil || rg.RUSettings.RU.Settings == nil {
```
Comment (JmPotato) [inline pkg/mcs/resource_manager/server/resource_group.go:170]: Should we check whether `rg.RUSettings.RU` is nil?

**Review by JmPotato — COMMENTED** [review]
The rest LGTM.

File: pkg/mcs/resource_manager/server/resource_group.go (hunk 1 above)
Comment (CabinfeverB) [reply pkg/mcs/resource_manager/server/resource_group.go:170]: `rg.RUSettings.RU` is not a pointer

**Review by JmPotato — APPROVED** [review]

**Pull request merged**